### ISSUE TITLE
Consul Catalog stickiness Does Not Work With Prefixes

### ISSUE THREAD

**Issue by charlieoleary** [containous/traefik#2623]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

- Operation cluster with Consul, Registrator, and Traefik.
- Registered containers with appropriate Traefik labels:

```
stage.traefik.backend.loadbalancer.stickiness.cookieName=_pin
stage.traefik.backend.loadbalancer.stickiness=true
stage.traefik.frontend.priority=10
stage.traefik.frontend.rule=PathPrefix:/api
stage.traefik.enable=true
``` 
- Ran Traefik with the following config:

```shell
./traefik \
    --debug=true \
    --configFile=config.toml \
    --consulcatalog \
    --consulcatalog.prefix=stage.traefik \
    --consulcatalog.debugloggeneratedtemplate \
    --consulcatalog.exposedbydefault=false \
    --consulcatalog.endpoint=127.0.0.1:8500
```

### What did you expect to see?

```toml
  [backends."backend-client".loadbalancer]
    method = "wrr"
    sticky = false

    [backends."backend-client".loadbalancer.stickiness]
      cookieName = "_pin"
```

### What did you see instead?

```toml
  [backends."backend-client".loadbalancer]
    method = "wrr"
    sticky = false
```

**Note:** This works as intended if I simply use the standard `traefik` prefix. Setting the `prefix` to `stage.traefik` makes stickiness stop functioning.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.4.3
Codename:     roquefort
Go version:   go1.9.2
Built:        2017-11-14_11:33:45AM
OS/Arch:      linux/amd64
```

(Also tested on 1.5.0-rc3)

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
defaultEntryPoints = ["http"]

[entryPoints]
  [entryPoints.http]
    address = ":8080"
    compress = true

[web]
  address = ":8585"
```

```shell
./traefik \
    --debug=true \
    --configFile=config.toml \
    --consulcatalog \
    --consulcatalog.prefix=stage.traefik \
    --consulcatalog.debugloggeneratedtemplate \
    --consulcatalog.exposedbydefault=false \
    --consulcatalog.endpoint=127.0.0.1:8500
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```console
Configuration received from provider consul_catalog: {"backends":{"backend-client":{"servers":{"client--172-nn-nn-nn--32873--version-build-v1-6--environment-stage--stage-traefik-backend-loadbalancer-stickiness-cookieName-pin--stage-traefik-backend-loadbalancer-stickiness-true--stage-traefik-frontend-priority-10--stage-traefik-frontend-rule-PathPrefix-api--stage-traefik-enable-true--0":{"url":"http://172.nn.nn.nn:32873","weight":0},"client--172-nn-nn-nn--32874--version-build-v1-6--environment-stage--stage-traefik-backend-loadbalancer-stickiness-cookieName-pin--stage-traefik-backend-loadbalancer-stickiness-true--stage-traefik-frontend-priority-10--stage-traefik-frontend-rule-PathPrefix-api--stage-traefik-enable-true--1":{"url":"http://172.nn.nn.nn:32874","weight":0},"client--172-nn-nn-nn--32875--version-build-v1-6--environment-stage--stage-traefik-backend-loadbalancer-stickiness-cookieName-pin--stage-traefik-backend-loadbalancer-stickiness-true--stage-traefik-frontend-priority-10--stage-traefik-frontend-rule-PathPrefix-api--stage-traefik-enable-true--2":{"url":"http://172.nn.nn.nn:32875","weight":0},"client--172-nn-nn-nn--32876--version-build-v1-6--environment-stage--stage-traefik-backend-loadbalancer-stickiness-cookieName-pin--stage-traefik-backend-loadbalancer-stickiness-true--stage-traefik-frontend-priority-10--stage-traefik-frontend-rule-PathPrefix-api--stage-traefik-enable-true--3":{"url":"http://172.nn.nn.nn:32876","weight":0},"client--172-nn-nn-nn--32877--version-build-v1-6--environment-stage--stage-traefik-backend-loadbalancer-stickiness-cookieName-pin--stage-traefik-backend-loadbalancer-stickiness-true--stage-traefik-frontend-priority-10--stage-traefik-frontend-rule-PathPrefix-api--stage-traefik-enable-true--4":{"url":"http://172.nn.nn.nn:32877","weight":0},"client--172-nn-nn-nn--32878--version-build-v1-6--environment-stage--stage-traefik-backend-loadbalancer-stickiness-cookieName-pin--stage-traefik-backend-loadbalancer-stickiness-true--stage-traefik-frontend-priority-10--stage-traefik-frontend-rule-PathPrefix-api--stage-traefik-enable-true--5":{"url":"http://172.nn.nn.nn:32878","weight":0}},"loadBalancer":{"method":"wrr"}}}}
DEBU[2017-12-27T05:21:18Z] Last consul_catalog config received less than %!s(flaeg.Duration=2000000000), waiting...
DEBU[2017-12-27T05:21:20Z] Waited for consul_catalog config, OK
DEBU[2017-12-27T05:21:20Z] Creating frontend traefik-admin
DEBU[2017-12-27T05:21:20Z] Wiring frontend traefik-admin to entryPoint http
DEBU[2017-12-27T05:21:20Z] Creating load-balancer wrr
DEBU[2017-12-27T05:21:20Z] Creating server server1 at http://localhost:8585 with weight 10
DEBU[2017-12-27T05:21:20Z] Creating frontend frontend-client
DEBU[2017-12-27T05:21:20Z] Wiring frontend frontend-client to entryPoint http
DEBU[2017-12-27T05:21:20Z] Creating route route-host-client PathPrefix:/api
DEBU[2017-12-27T05:21:20Z] Creating backend backend-client
DEBU[2017-12-27T05:21:20Z] Creating load-balancer wrr
```

(Sanitized...)

**Comment by ldez** [comment]
I'm don't understand: your are using Docker labels or Consul KV?

- Labels on containers are for Docker and don't support prefix.
- The prefix is used for tags and keys in Consul

**Comment by charlieoleary** [comment]
Yes, sorry for any confusion. I'm using Consul Catalog and Registrator to register services with the Consul catalog. 

Just to further clarify, if I register with the standard `traefik.<labelSuffix>` everything works as expected, including stickies. If I change the prefix to `stage.traefik.<labelSuffix>` and set the `--consulcatalog.prefix=stage.traefik` everything works EXCEPT the stickiness setting.

**Comment by charlieoleary** [comment]
Amazing, @ldez! Thanks so much for jumping on this! 🎉

**Comment by traefiker** [comment]
Closed by #2624.
